interactors/nonceHandlerV3/nonceTransactionsHandler_test.go

```diff
@@ -231,6 +233,173 @@ func TestSendTransactionsCloseDelay(t *testing.T) {
 	require.NoError(t, err, "failed to send transactions as bulk")
 }
 
+func TestApplyNonceAndGasPriceConcurrently(t *testing.T) {
+	t.Parallel()
+
+	var getAccountCalled bool
+
+	transactionHandler, err := NewNonceTransactionHandlerV3(createMockArgsNonceTransactionsHandlerV3(&getAccountCalled))
+	require.NoError(t, err, "failed to create transaction handler")
+
+	var txs []*transaction.FrontendTransaction
+
+	for i := 0; i < 100; i++ {
+		tx := &transaction.FrontendTransaction{
+			Sender:   testAddressAsBech32String,
+			Receiver: testAddressAsBech32String,
+			GasLimit: 50000,
+			ChainID:  "T",
+			Value:    "5000000000000000000",
+			Nonce:    uint64(i),
+			GasPrice: 1000000000,
+			Version:  2,
+		}
+		txs = append(txs, tx)
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(1)
```

Comment:
L261-L304 can be done in a for loop.
Also, no need to panic the code in the go routine, can use a require.NotNil for the inner err

Reply:
done